### PR TITLE
Improve PHPStan use and explanation in Readme 

### PR DESCRIPTION
**Pull request by Quetzacoalt91**

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -427,6 +427,50 @@
             ],
             "time": "2018-02-15T16:58:55+00:00"
         },
+        {
+            "name": "phpstan/phpstan-shim",
```
Comment (PierreRambaud) [inline composer.lock]: revert this change?
```
            "require": {
                "php": "~7.1"
            },
```

Comment (Quetzacoalt91) [reply composer.lock]: True, I ran `composer update`

File: composer.lock [hunk 2]
```diff
@@ -110,34 +110,34 @@
...
-                "url": "https://api.github.com/repos/doctrine/annotations/zipball/54cacc9b81758b14e3ce750f205a393d52339e97",
-                "reference": "54cacc9b81758b14e3ce750f205a393d52339e97",
+                "url": "https://api.github.com/repos/doctrine/annotations/zipball/53120e0eb10355388d6ccbe462f1fea34ddadb24",
+                "reference": "53120e0eb10355388d6ccbe462f1fea34ddadb24",
                 "shasum": ""
             },
             "require": {
                 "doctrine/lexer": "1.*",
-                "php": "^5.6 || ^7.0"
+                "php": "^7.1"
```
Comment (PierreRambaud) [inline composer.lock]: You need to revert your change again :/

File: composer.lock [hunk 3]
```diff
@@ -329,33 +329,29 @@
         },
         {
             "name": "paragonie/random_compat",
-            "version": "v2.0.18",
+            "version": "v9.99.99",
```
Comment (PierreRambaud) [inline composer.lock]: lol?

File: composer.lock [hunk 4]
```diff
@@ -329,33 +329,29 @@
...
                 "type": "zip",
-                "url": "https://api.github.com/repos/paragonie/random_compat/zipball/0a58ef6e3146256cc3dc7cc393927bcc7d1b72db",
-                "reference": "0a58ef6e3146256cc3dc7cc393927bcc7d1b72db",
+                "url": "https://api.github.com/repos/paragonie/random_compat/zipball/84b4dfb120c6f9b4ff7b3685f9b8f1aa365a0c95",
+                "reference": "84b4dfb120c6f9b4ff7b3685f9b8f1aa365a0c95",
                 "shasum": ""
             },
             "require": {
-                "php": ">=5.2.0"
+                "php": "^7"
```
Comment (PierreRambaud) [inline composer.lock]: Same here

File: README.md [hunk 5]
```diff
@@ -9,22 +9,53 @@ This repository provides a configuration for [friendsofphp/php-cs-fixer](http://
...
+$ php vendor/bin/prestashop-coding-standards cs-fixer:init [--dest /path/to/my/project]
 ```
 
 It'll create a configuration file `.php_cs.dist` in the root of your project.
 
 ### Phpstan
 
 ```bash
-$ ./bin/prestashop-coding-standards phptan:init [--dest /path/to/my/project]
+$ php vendor/bin/prestashop-coding-standards phptan:init [--dest /path/to/my/project]
```
Comment (mickaelandrieu) [inline README.md]: phptan or phpstan ?

File: README.md [hunk 6]
```diff
@@ -9,22 +9,53 @@ This repository provides a configuration for [friendsofphp/php-cs-fixer](http://
...
 ```
 
-It'll create a `boostrap.php` and a `phpstan.neon` files, that are required to run phpstan.
+It'll create a `boostrap.php` and a `phpstan.neon` files, by default in `tests/phpstan`, that are required to run phpstan.
```
Comment (mickaelandrieu) [inline README.md]:
```diff
- boostrap.php
+ bootstrap.php
```

File: README.md [hunk 7]
```diff
@@ -9,22 +9,53 @@ This repository provides a configuration for [friendsofphp/php-cs-fixer](http://
...
 The default phpstan level is the lowest available, but we recommend you to update this value to get more recommandations.
 
+PHPStan is not provided by our dependancies, because of the PHP compatibility from projects using this repository. We recommend you to install it globally on your environment:
```
Comment (mickaelandrieu) [inline README.md]:
```diff
- dependancies
+ dependencies
```

File: README.md [hunk 8]
```diff
@@ -9,22 +9,53 @@ This repository provides a configuration for [friendsofphp/php-cs-fixer](http://
...
+
+### PHP CS Fixer
+
+```php
+php vendor/bin/php-cs-fixer fix
+```
+
+### PHPStan
+
+If you installed PHPStan globally and made the folder available in your PATH:
```
Comment (mickaelandrieu) [inline README.md]:
```diff
- If you installed
+ If you have installed
```

File: README.md (hunk 5 above)
Comment (Quetzacoalt91) [reply README.md]: Nothing related to the sun :sunglasses:

File: templates/phpstan/bootstrap.php [hunk 9]
```diff
@@ -18,8 +20,14 @@
...
 // Make sure loader php-parser is coming from php stan composer
+
+// 1- Use with Docker container
+$loader = new \Composer\Autoload\ClassLoader();
+$loader->setPsr4('PhpParser\\', ['/composer/vendor/nikic/php-parser/lib/PhpParser']);
+$loader->register(true);
+// 2- Use with PHPStan phar
 $loader = new \Composer\Autoload\ClassLoader();
-$loader->setPsr4('PhpParser\\', array('/composer/vendor/nikic/php-parser/lib/PhpParser'));
+$loader->setPsr4('PhpParser\\', ['phar://' . $pathToModuleRoot . 'vendor/phpstan/phpstan-shim/phpstan.phar/vendor/nikic/php-parser/lib/PhpParser/']);
```
Comment (mickaelandrieu) [inline templates/phpstan/bootstrap.php]: this is smart :)

Comment (Quetzacoalt91) [reply templates/phpstan/bootstrap.php]: Yeah, but I just remember that I had to remove this from the dependencies. It won't work anymore.
